components/workspace-rollout-job/pkg/rollout/rollout.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package rollout
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	logrus "github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/workspace-rollout-job/pkg/analysis"
+	"github.com/gitpod-io/gitpod/workspace-rollout-job/pkg/wsbridge"
+)
+
+type RollOutJob struct {
+	oldCluster           string
+	newCluster           string
+	currentScore         int32
+	okayScoreUntilNoData int32
+	analyzer             analysis.Analyzer
+	RolloutAction        wsbridge.RolloutAction
+	rolloutStep          int32
+	analysisWaitDuration time.Duration
+
+	ticker *time.Ticker
+	revert chan bool
+	done   chan bool
+}
+
+func New(oldCluster, newCluster string, rolloutWaitDuration, analysisWaitDuration time.Duration, step, okayScoreUntilNoData int32, analyzer analysis.Analyzer, rolloutAction wsbridge.RolloutAction) *RollOutJob {
+	return &RollOutJob{
+		oldCluster:           oldCluster,
+		newCluster:           newCluster,
+		okayScoreUntilNoData: okayScoreUntilNoData,
+		currentScore:         0,
+		rolloutStep:          step,
+		analyzer:             analyzer,
+		RolloutAction:        rolloutAction,
+		done:                 make(chan bool, 1),
+		revert:               make(chan bool, 1),
+		analysisWaitDuration: analysisWaitDuration,
+		// move forward every waitDuration
+		ticker: time.NewTicker(rolloutWaitDuration),
+	}
+}
+
+// Start runs the job synchronously
+func (r *RollOutJob) Start(ctx context.Context) error {
+	log := logrus.WithField("component", "rollout-job")
```

Comment:
I believe all components from Gitpod use our own common-go/log library for our logs. Would it make sense to use it as well just to make log format consistent with all other Gitpod components?

Reply:
Updated! 👍🏼